src/main/java/org/folio/orders/events/handlers/ReceiptStatusConsistency.java

```diff
@@ -0,0 +1,195 @@
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.rest.acq.model.Piece;
+import org.folio.rest.acq.model.Piece.ReceivingStatus;
+import org.folio.rest.acq.model.PieceCollection;
+import org.folio.rest.impl.AbstractHelper;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.jaxrs.model.PoLine.ReceiptStatus;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+import one.util.streamex.StreamEx;
+
+@Component("receiptStatusHandler")
+public class ReceiptStatusConsistency extends AbstractHelper implements Handler<Message<JsonObject>> {
+
+  private static final int LIMIT = 500;
+  private static final int OFFSET = 0;
+  private static final String PIECES_ENDPOINT = resourcesPath(PIECES) + "?query=poLineId==%s&lang=%s";
+  private static final String PIECE_ID_UPDATE = "pieceIdUpdate";
+  private static final String RECEIVING_STATUS_BEFORE_UPDATE = "receivingStatusBeforeUpdate";
+  
+  @Autowired
+  public ReceiptStatusConsistency(Vertx vertx) {
+    super(vertx.getOrCreateContext());
+  }
+
+  @Override
+  public void handle(Message<JsonObject> message) {
+    JsonObject body = message.body();
+    String lang = body.getString(LANG);
+
+    logger.info("Received message body: {}", body);
+
+    Map<String, String> okapiHeaders = org.folio.orders.utils.HelperUtils.getOkapiHeaders(message);
+    HttpClientInterface httpClient = getHttpClient(okapiHeaders, true);
+
+    String pieceIdUpdate = body.getString(PIECE_ID_UPDATE);
+
+    List<CompletableFuture<String>> futures = new ArrayList<>();
+    CompletableFuture<String> future = new VertxCompletableFuture<>(ctx);
+
+    futures.add(future);
+
+    // 1. Get updated piece by id from storage
+    getPieceById(pieceIdUpdate, lang, httpClient, ctx, okapiHeaders, logger).thenAccept(jsonPiece -> {
+      Piece piece = jsonPiece.mapTo(Piece.class);
+      logger.info("piece for pieceRecord.json" + piece + " pieceById: " + pieceIdUpdate);
+      ReceivingStatus receivingStatus = piece.getReceivingStatus();
+      String receivingStatusBeforeUpdate = body.getString(RECEIVING_STATUS_BEFORE_UPDATE);
+
+      // 2. if receivingStatus is different - before writing to storage vs after writing to storage
+      if (!receivingStatus.toString()
+        .equalsIgnoreCase(receivingStatusBeforeUpdate)) {
+        String poLineId = piece.getPoLineId();
+
+        String endpoint = String.format(PIECES_ENDPOINT, poLineId, LANG);
+
+        // 3. Get all pieces for poLineId above
+        getPieces("", endpoint, httpClient, ctx, okapiHeaders, logger).thenAccept(piecesCollection -> {
```

Comment:
Why you bake the query into the endpoint above and pass in an empty string for query here?

Reply:
Done, the refactored code to avoid passing an empty string for the query.